grid_map_visualization/src/GridMapVisualization.cpp

```diff
@@ -6,165 +6,170 @@
-    auto visualization = factory_.getInstance(type, name);
-    visualization->readParameters(config[i]);
+    auto visualization = factory_->getInstance(type, name);
+    visualization->readParameters();
     visualizations_.push_back(visualization);
-    ROS_INFO("%s: Configured visualization of type '%s' with name '%s'.",
-             visualizationsParameter_.c_str(), type.c_str(), name.c_str());
+    RCLCPP_INFO(
+      node_->get_logger(), "%s: Configured visualization of type '%s' with name '%s'.",
+      visualizationsParameter_.c_str(), type.c_str(), name.c_str());
   }
-
   return true;
 }
 
 bool GridMapVisualization::initialize()
 {
-  for (auto& visualization : visualizations_) {
+  for (auto visualization : visualizations_) {
     visualization->initialize();
   }
-  updateSubscriptionCallback(ros::TimerEvent());
-  ROS_INFO("Grid map visualization initialized.");
+
+  updateSubscriptionCallback();
+  RCLCPP_INFO(node_->get_logger(), "Grid map visualization initialized.");
   return true;
 }
 
-void GridMapVisualization::updateSubscriptionCallback(const ros::TimerEvent&)
+void GridMapVisualization::updateSubscriptionCallback()
 {
   bool isActive = false;
 
-  for (auto& visualization : visualizations_) {
+  for (auto & visualization : visualizations_) {
     if (visualization->isActive()) {
       isActive = true;
       break;
     }
   }
 
   if (!isSubscribed_ && isActive) {
-    mapSubscriber_ = nodeHandle_.subscribe(mapTopic_, 1, &GridMapVisualization::callback, this);
+    mapSubscriber_ = node_->create_subscription<grid_map_msgs::msg::GridMap>(
+      mapTopic_, rclcpp::SystemDefaultsQoS(),
+      std::bind(&GridMapVisualization::callback, this, std::placeholders::_1));
+
     isSubscribed_ = true;
-    ROS_DEBUG("Subscribed to grid map at '%s'.", mapTopic_.c_str());
+    RCLCPP_DEBUG(node_->get_logger(), "Subscribed to grid map at '%s'.", mapTopic_.c_str());
   }
   if (isSubscribed_ && !isActive) {
-    mapSubscriber_.shutdown();
+    mapSubscriber_.reset();
     isSubscribed_ = false;
-    ROS_DEBUG("Cancelled subscription to grid map.");
+    RCLCPP_DEBUG(node_->get_logger(), "Cancelled subscription to grid map.");
   }
 }
 
-void GridMapVisualization::callback(const grid_map_msgs::GridMap& message)
+void GridMapVisualization::callback(const grid_map_msgs::msg::GridMap::SharedPtr message)
 {
-  ROS_DEBUG("Grid map visualization received a map (timestamp %f) for visualization.",
-            message.info.header.stamp.toSec());
+  RCLCPP_DEBUG(
+    node_->get_logger(),
+    "Grid map visualization received a map (timestamp %f) for visualization.",
+    rclcpp::Time(message->header.stamp).seconds());
   grid_map::GridMap map;
-  grid_map::GridMapRosConverter::fromMessage(message, map);
+  grid_map::GridMapRosConverter::fromMessage(*message, map);
 
-  for (auto& visualization : visualizations_) {
-    visualization->visualize(map);
+  for (std::size_t i = 0; i < visualizations_.size(); i++) {
```

Comment:
Did it not work the other way around?

Reply:
Will reverted this change, removed while debugging and forgot to revert it back, my bad.